rush-plugins/rush-amazon-s3-build-cache-plugin/src/RushAmazonS3BuildCachePlugin.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import { Import, JsonSchema } from '@rushstack/node-core-library';
+import { IRushPlugin, RushSession, RushConfiguration } from '@microsoft/rush-lib';
+
+const AmazonS3BuildCacheProviderModule: typeof import('./AmazonS3BuildCacheProvider') = Import.lazy(
+  '../logic/buildCache/AmazonS3/AmazonS3BuildCacheProvider',
+  require
+);
+import type { AmazonS3BuildCacheProvider } from './AmazonS3BuildCacheProvider';
+
+const PLUGIN_NAME: string = 'AmazonS3BuildCachePlugin';
+
+/**
+ * @public
+ */
+export interface IAmazonS3ConfigurationJson {
+  /**
+   * The Amazon S3 region of the bucket to use for build cache (e.g. "us-east-1").
+   */
+  s3Region: string;
+
+  /**
+   * The name of the bucket in Amazon S3 to use for build cache.
+   */
+  s3Bucket: string;
+
+  /**
+   * An optional prefix ("folder") for cache items.
+   */
+  s3Prefix?: string;
+
+  /**
+   * If set to true, allow writing to the cache. Defaults to false.
+   */
+  isCacheWriteAllowed?: boolean;
+}
+
+/**
+ * @public
+ */
+export class RushAmazonS3BuildCachePlugin implements IRushPlugin {
+  public pluginName: string = PLUGIN_NAME;
+
+  private static _jsonSchema: JsonSchema = JsonSchema.fromFile(
+    path.join(__dirname, 'schemas', 'amazon-s3-config.schema.json')
```

Comment:
Can we please lazy-load this as well?

Reply:
Ideally plugins should require no imports whatsoever (except type-only imports that get erased) until their taps get invoked.

Reply:
Okay, it's essential to make a high standard on official plugins. Maybe boilerplates/notes should be documented to help third-party plugins